fixing pkg_config empty dirs

memsharded opened the pull request:
Changelog: Bugfix: Fix ``pkg_config`` generator adding to .pc files empty include and lib dirs.
Docs: Omit

Fix https://github.com/conan-io/conan/issues/7701

memsharded committed:
fixing pkg_config empty dirs

memsharded committed:
fixing tests

SSE4 reviewed (approved):
LGTM. maybe we also should filter framework paths, but I guess it's out of the scope.

jgsogo reviewed (changes requested):
Why all the `filter_empty = False`? That flag removes the paths that do not exist, but it doesn't turn an empty list `[]` into `None`, so the change should be useless in the context of this PR.

memsharded commented:
> Why all the filter_empty = False? That flag removes the paths that do not exist, but it doesn't turn an empty list [] into None, so the change should be useless in the context of this PR.

No, all those tests are broken otherwise. They are defining empty folder, that are being added to pkg_config despite being empty, and the default being ``filter_empty=True``. We need to either change the output (the .pc) checks, I thought it was more readable the change to define just the ``filter_empty=False``.

Anyway, the approach is not valid anyway, I need to fix it. Will review and update this PR, please tell me if you prefer the output of the tests changed instead of the filter_empty.

memsharded commented on conans/client/generators/pkg_config.py:
```diff
@@ -97,16 +97,19 @@ def single_pc_file_contents(self, name, cpp_info, comp_requires_gennames, is_com
-        if dir_lines:
-            libdir_vars = varnames
-            lines.extend(dir_lines)
+        if cpp_info.lib_paths:  # Make sure it is not empty
```
This is not valid. This can still add empty dirs to the .pc files.

memsharded committed:
Merge branch 'release/1.29' into fix/pkg_config_empty_dirs

jgsogo commented:
Better to change the output. Not sure if the `filter_empty` is something we want to keep as an attribute in the `cpp_info` object or closer to the place where the information is written/consumed (I think this second approach would be better).

memsharded committed:
fixed pkg_config adding paths only if exists

memsharded committed:
review: improve functional tests

memsharded committed:
fixing tests

memsharded commented:
I have changed some tests, others were very difficult or almost impossible to make it work without ``filter_empty=False``, for example when using absolute paths, as it is not possible to write in / for testing. So I have changed some of them, forcing the addition of files inside the folders, so they are not empy and the generated .pc files contain the information of folders.
